pyvrp/cpp/CostEvaluator.h

```diff
@@ -23,6 +23,51 @@ template <typename T> concept CostEvaluatable = requires(T arg)
     // clang-format on
 };
 
+/**
+ * RouteData(
+ *     size: int,
+ *     distance: Distance,
+ *     load: Load,
+ *     timeWarp: Duration
+ * )
+ *
+ * Simple data object storing route data as properties.
+ *
+ * Parameters
+ * ----------
+ * size
+ *     Size of the route.
+ * distance
+ *     Distance traveled in the route.
+ * load
+ *     Load carried in the route.
+ * timeWarp
+ *     Time warp for the route.
+ *
+ * Attributes
+ * ----------
+ * size
+ *     Size of the route.
+ * distance
+ *     Distance traveled in the route.
+ * load
+ *     Load carried in the route.
+ * timeWarp
+ *     Time warp for the route.
+ */
+struct RouteData
```

Comment:
Note: `RouteData` doesn't implement `CostEvaluatable` concept as it does not have `uncollectedPrizes`, `fixedVehicleCost`, `loadExcess`. While we could change that (or change the concept) I think it is cleaner this way: the concept is about the cost structure for a solution whereas this object is for a route. We could consider moving away from the concept (as we use it only once now) and also implement that based on a similar `SolutionData` (or properties or better name) that tracks KPI's of a solution.

Reply:
Like `pyvrp.search.Route` now does, I think this new object could also implement the member functions related to the concept. Then no further specialisation is necessary.